File: srcr/src/days_2025/day3.rs
pub fn solve_day_3(input: &str) -> u64 {
    const TARGET: usize = 12;

    let total = input.lines()
        .filter_map(|line| {
            let chars: Vec<char> = line.chars().collect();
            let n = chars.len();

            let mut selected = Vec::with_capacity(TARGET);
            let mut start = 0;

            for pos in 0..TARGET {
                let window_end = n - (TARGET - pos - 1);

                let mut best_digit = '0';
                let mut best_idx = start;

                for i in start..window_end {
                    if chars[i] > best_digit {
                        best_digit = chars[i];
                        best_idx = i;
                    }
                }

                selected.push(best_digit);
                start = best_idx + 1;
            }

            let joltage = selected.iter()
                .fold(0u64, |acc, &ch| acc * 10 + ch.to_digit(10).unwrap() as u64);

            Some(joltage)
        })
        .sum();

    println!("Total : {}", total);
    total
}

File: srcd/bin/2025/day3.dart

class Day3_2025{
  final String input;

  Day3_2025._({
    required this.input
  });

  factory Day3_2025(String input) {
    return Day3_2025._(input: input);
  }

  int getResult(int target) {

    final List<String> lines = input.split("\n");

    int total = lines.map((line) {
      final List<String> chars = line.split('');
      final int n = chars.length;

      final List<String> selected = [];
      int start = 0;

      for (int pos = 0; pos < target; pos++) {
        final int windowEnd = n - (target - pos - 1);

        String bestDigit = '0';
        int bestIdx = start;

        for (int i = start; i < windowEnd; i++) {
          if (chars[i].compareTo(bestDigit) > 0) {
            bestDigit = chars[i];
            bestIdx = i;
          }
        }

        selected.add(bestDigit);
        start = bestIdx + 1;
      }

      final int joltage = selected.fold(0, (acc, ch) {
        return acc * 10 + int.parse(ch);
      });

      return joltage;
    }).fold(0, (a, b) => a + b);

    print('Total : $total');
    return total;
  }

}